Add "making test data files" section to contributing guide (#233)
hyperspy/rosettasciio: upcoming_changes/233.enhancements.rst

> Add :ref:`making test data files <making_test_files>` section to contributing guide, explain what are "good" test data files and the reasoning.

Reviewer: ```suggestion
Add :ref:`making test data files <making_test_files>` section to contributing guide, explain characteristics of "good" test data files.
```
